lib/turbograft/xhr_headers.rb

```diff
@@ -31,7 +31,7 @@ def store_for_turbolinks(url)
       end
 
       def set_xhr_redirected_to
-        if session && session[:_turbolinks_redirect_to]
+        if session && session[:_turbolinks_redirect_to] && session[:_turbolinks_redirect_to] == request.original_url
```

Comment:
Though I think that `original_request` is probably misleading, we're looking for the request that is pointed where we told the first one to redirect to. It would be more like

`request_matches_redirect?`